tools/autograd/gen_variable_type.py

```diff
@@ -826,6 +833,10 @@ def emit_fw_derivatives() -> List[str]:
                                        'formula has been defined for it. This case should only happen for function that '
                                        'take a single TensorList as input. All other cases are not supported right now.')
                 requires_fw_grad = "true"
+            if fn.info and fn.info.output_differentiability_conditions:
+                assert len(fn.info.output_differentiability_conditions) == 1
+                requires_fw_grad = \
+                    f'({requires_fw_grad}) && ({fn.info.output_differentiability_conditions[0]})'
```

Comment:
I think you want the extra condition first. Because you don't want to try and access forward grads for Tensors that you shouldn't.